tldw_Server_API/tests/RAG/test_query_rewriting_loop.py

```
@@ -0,0 +1,416 @@
+            strategies=["improve_for_retrieval"],
+            failed_docs=sample_failed_docs,
+            failure_reason="avg_relevance_0.2",
+        )
+
+        # Should generate rewrites
+        assert len(rewrites) > 0
+
+        # At least one rewrite should incorporate info from failed docs
+        rewrite_texts = [r.rewritten_query for r in rewrites]
+        # Check if any entity from docs was incorporated
+        has_entity_based = any(
```

Ruff (0.14.14)

[error] Local variable `has_entity_based` is assigned to but never used

Remove assignment to unused variable `has_entity_based`

(F841)